lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp

```diff
@@ -2131,29 +2131,32 @@ lldb::TypeSystemSP SwiftASTContext::CreateInstance(lldb::LanguageType language,
     handled_sdk_path = true;
   }
 
-  if (target.GetSwiftCreateModuleContextsInParallel()) {
-    // The first call to GetTypeSystemForLanguage() on a module will
-    // trigger the import (and thus most likely the rebuild) of all
-    // the Clang modules that were imported in this module. This can
-    // be a lot of work (potentially ten seconds per module), but it
-    // can be performed in parallel.
-    const unsigned threads =
-        repro::Reproducer::Instance().IsReplaying() ? 1 : 0;
-    llvm::ThreadPool pool(llvm::hardware_concurrency(threads));
-    for (size_t mi = 0; mi != num_images; ++mi) {
-      auto module_sp = target.GetImages().GetModuleAtIndex(mi);
-      pool.async([=] {
-        auto val_or_err =
-            module_sp->GetTypeSystemForLanguage(lldb::eLanguageTypeSwift);
-        if (!val_or_err) {
-          llvm::consumeError(val_or_err.takeError());
-        }
-      });
+  auto warmup_astcontexts = [&]() {
+    if (target.GetSwiftCreateModuleContextsInParallel()) {
+      // The first call to GetTypeSystemForLanguage() on a module will
+      // trigger the import (and thus most likely the rebuild) of all
+      // the Clang modules that were imported in this module. This can
+      // be a lot of work (potentially ten seconds per module), but it
+      // can be performed in parallel.
+      const unsigned threads =
+          repro::Reproducer::Instance().IsReplaying() ? 1 : 0;
+      llvm::ThreadPool pool(llvm::hardware_concurrency(threads));
+      for (size_t mi = 0; mi != num_images; ++mi) {
+        auto module_sp = target.GetImages().GetModuleAtIndex(mi);
+        pool.async([=] {
+          auto val_or_err =
+              module_sp->GetTypeSystemForLanguage(lldb::eLanguageTypeSwift);
+          if (!val_or_err) {
+            llvm::consumeError(val_or_err.takeError());
```

Comment:
what kind of errors might these be? I am wondering whether we should log these.

Reply:
If I read `llvm::Expected<TypeSystem &> TypeSystemMap::GetTypeSystemForLanguage(lldb::LanguageType language,
                                        Module *module, bool can_create) {` right, it never returns an error.